src/model/SubSurface.cpp

```diff
@@ -1595,6 +1596,37 @@ namespace model {
   void SubSurface::resetShadingControl() {
     removeAllShadingControls();
   }
+
+  std::vector<Point3d> SubSurface::roughOpeningVertices() const {
+    auto frameAndDivider = windowPropertyFrameAndDivider();
+    if (frameAndDivider) {
+      double fw = frameAndDivider->frameWidth();
+      // Get a transform to change the points to x/y
+      Transformation faceTransform = Transformation::alignFace(this->vertices());
+      std::vector<Point3d> faceVertices = faceTransform.inverse() * this->vertices();
+      // Offset the points by the framewidth
+      boost::optional<std::vector<Point3d>> offset = openstudio::buffer(faceVertices, fw, 0.01);
+      if (!offset) {
+        // If offset failed it is because the points are in the wrong order
+        // If it fails again then something went awry with boost::buffer
+        faceVertices = openstudio::reverse(faceVertices);
+        offset = openstudio::buffer(faceVertices, fw, 0.01);
+        std::vector<Point3d> roughOpeningVertices = faceTransform * offset.get();
+        return roughOpeningVertices;
+      }
+    }
```

Comment:
Should it be this instead?

```suggestion
      boost::optional<std::vector<Point3d>> offset = openstudio::buffer(faceVertices, fw, 0.01);
      if (!offset) {
        // If offset failed it is because the points are in the wrong order
        // If it fails again then something went awry with boost::buffer
        faceVertices = openstudio::reverse(faceVertices);
        offset = openstudio::buffer(faceVertices, fw, 0.01);
      }
      std::vector<Point3d> roughOpeningVertices = faceTransform * offset.get();
      return roughOpeningVertices;
    }
```